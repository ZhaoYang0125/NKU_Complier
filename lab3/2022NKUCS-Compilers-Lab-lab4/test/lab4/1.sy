int a;

int main()
{
    int a;
    a = 0xF + 02+ 0xf;
    if(a <= 5)
        return 1;
    return 0;
}