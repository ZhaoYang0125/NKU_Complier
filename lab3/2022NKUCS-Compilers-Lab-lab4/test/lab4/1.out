[DEBUG LAB4]: 	INT   	int	1	0	
[DEBUG LAB4]: 	ID    	a	1	4	0x5567bdb42eb0
[DEBUG LAB4]: 	SEMICOLON	;	1	5	
[DEBUG LAB4]: 	INT   	int	3	0	
[DEBUG LAB4]: 	MAIN  	main	3	4	
[DEBUG LAB4]: 	LPAREN	(	3	8	
[DEBUG LAB4]: 	RPAREN	)	3	9	
[DEBUG LAB4]: 	LBRACE	{	4	0	
[DEBUG LAB4]: 	INT   	int	5	4	
[DEBUG LAB4]: 	ID    	a	5	8	0x5567bdb49830
[DEBUG LAB4]: 	SEMICOLON	;	5	9	
[DEBUG LAB4]: 	ID    	a	6	4	0x5567bdb49830
[DEBUG LAB4]: 	ASSIGN	=	6	6	
[DEBUG LAB4]: 	HEXADECIMIAL	15	6	8	
[DEBUG LAB4]: 	ADD   	+	6	9	
[DEBUG LAB4]: 	OCTAL 	2	6	13	
[DEBUG LAB4]: 	ADD   	+	6	13	
[DEBUG LAB4]: 	HEXADECIMIAL	15	6	15	
[DEBUG LAB4]: 	SEMICOLON	;	6	15	
[DEBUG LAB4]: 	IF    	if	7	4	
[DEBUG LAB4]: 	LPAREN	(	7	6	
[DEBUG LAB4]: 	ID    	a	7	7	0x5567bdb49830
[DEBUG LAB4]: 	LEQUAL	<=	7	9	
[DEBUG LAB4]: 	DECIMIAL	5	7	12	
[DEBUG LAB4]: 	RPAREN	)	7	13	
[DEBUG LAB4]: 	RETURN	return	8	8	
[DEBUG LAB4]: 	DECIMIAL	1	8	15	
[DEBUG LAB4]: 	SEMICOLON	;	8	16	
[DEBUG LAB4]: 	RETURN	return	9	4	
[DEBUG LAB4]: 	DECIMIAL	0	9	11	
[DEBUG LAB4]: 	SEMICOLON	;	9	12	
[DEBUG LAB4]: 	RBRACE	}	10	0	
